core/src/banking_stage.rs

```diff
@@ -2108,4 +2098,177 @@ mod tests {
+            assert_eq!(packets[0].packets.len(), num_conflicting_transactions);
+            let all_packets = packets.pop().unwrap();
+            let mut buffered_packets: UnprocessedPackets = vec![(
+                all_packets,
+                (0..num_conflicting_transactions).into_iter().collect(),
+            )]
+            .into_iter()
+            .collect();
+
+            let (gossip_vote_sender, _gossip_vote_receiver) = unbounded();
+
+            // When the working bank in poh_recorder is None, no packets should be processed
+            BankingStage::consume_buffered_packets(
+                &Pubkey::default(),
+                &poh_recorder,
+                &mut buffered_packets,
+                None,
+                &gossip_vote_sender,
+                None::<Box<dyn Fn()>>,
+            );
+            assert_eq!(buffered_packets[0].1.len(), num_conflicting_transactions);
+
+            // When the poh recorder has a bank, should process all non conflicting buffered packets.
+            // Processes one packet per iteration of the loop
+            for i in 0..num_conflicting_transactions {
+                poh_recorder.lock().unwrap().set_bank(&bank);
+                BankingStage::consume_buffered_packets(
+                    &Pubkey::default(),
+                    &poh_recorder,
+                    &mut buffered_packets,
+                    None,
+                    &gossip_vote_sender,
+                    None::<Box<dyn Fn()>>,
+                );
+                let num_expected_unprocessed = num_conflicting_transactions - i - 1;
+                if num_expected_unprocessed == 0 {
+                    assert!(buffered_packets.is_empty())
+                } else {
+                    assert_eq!(buffered_packets[0].1.len(), num_expected_unprocessed);
+                }
+            }
+        }
+        Blockstore::destroy(&ledger_path).unwrap();
+    }
+
+    #[test]
+    fn test_consume_buffered_packets_interrupted() {
+        let ledger_path = get_tmp_ledger_path!();
+        {
+            let (transactions, bank, poh_recorder, _entry_receiver) =
+                setup_conflicting_transactions(&ledger_path);
+            let num_conflicting_transactions = transactions.len();
+            let packets = to_packets_chunked(&transactions, 1);
+            assert_eq!(packets.len(), num_conflicting_transactions);
+            for single_packets in &packets {
+                assert_eq!(single_packets.packets.len(), 1);
+            }
+            let mut buffered_packets: UnprocessedPackets = packets
+                .clone()
+                .into_iter()
+                .map(|single_packets| (single_packets, vec![0]))
+                .collect();
+
+            let (continue_sender, continue_receiver) = unbounded();
+            let (finished_packet_sender, finished_packet_receiver) = unbounded();
+
+            let test_fn = Some(move || {
+                finished_packet_sender.send(1).unwrap();
```

Comment:
it looks like actually anything is ok to `send()`, how about using the unit type?: `()`

Reply:
also, some naming is desired like `let control_flow_token = ();`.

Reply:
or `continuation_token`?

Reply:
:eyes:

Reply:
hmmm forgot to mention tried this and got this clippy error: ``#[warn(clippy::unit_arg)]`